src/shogun/mathematics/linalg/internal/implementation/Convolve.h

```diff
@@ -113,7 +114,7 @@ struct convolve<Backend::EIGEN3, Matrix>
 	 * @param stride_y Stride in the y (row) direction
 	 */
 	static void compute(SGMatrix<T> X, SGMatrix<T> W, SGMatrix<T> Y, bool flip ,
-		bool overwrite, int32_t stride_x, int32_t stride_y)
+		bool overwrite, int32_t stride_x, int32_t stride_y, ENLAutoencoderPosition autoencoder_position = NLAP_NONE)
```

Comment:
I see.

I support this idea in general. However, making the linalg::convolve interface depend on autoencoder parameters is not a good idea. I think you want to add a boolean flag and call it something else. Then I am fine with the idea. 

Would be good if @lisitsyn and @lambday could comment briefly